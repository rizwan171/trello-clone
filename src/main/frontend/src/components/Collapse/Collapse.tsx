import { useRef } from "react";
import { CollapseProps } from "../../types/components/CollapseProps";

const Collapse = ({ isOpen, children, className }: CollapseProps) => {
  const ref = useRef<HTMLDivElement>(null);
  const inlineStyle = isOpen ? { height: "100%" } : { height: 0, width: 0 };

  return (
    <div
      ref={ref}
      aria-hidden={!isOpen}
      style={inlineStyle}
      className={"transition-height ease overflow-hidden duration-500 " + className}
    >
      {children}
    </div>
  );
};

export default Collapse;