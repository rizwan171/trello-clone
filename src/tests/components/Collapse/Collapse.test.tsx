import { render } from "@testing-library/react";
import Collapse from "../../../components/Collapse/Collapse";

describe("Collapse", () => {
  it("should render successfully when open", () => {
    const view = render(
      <Collapse isOpen={true}>
        <div>test</div>
      </Collapse>
    );
    expect(view.asFragment()).toMatchSnapshot();
  });

  it("should render successfully when closed", () => {
    const view = render(
      <Collapse isOpen={false}>
        <div>test</div>
      </Collapse>
    );
    expect(view.asFragment()).toMatchSnapshot();
  });
});
