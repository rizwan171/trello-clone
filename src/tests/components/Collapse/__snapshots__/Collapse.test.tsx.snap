// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse should render successfully when closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="transition-height ease overflow-hidden duration-500 undefined"
    style="height: 0px; width: 0px;"
  >
    <div>
      test
    </div>
  </div>
</DocumentFragment>
`;

exports[`Collapse should render successfully when open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="transition-height ease overflow-hidden duration-500 undefined"
    style="height: 100%;"
  >
    <div>
      test
    </div>
  </div>
</DocumentFragment>
`;
